File: 2021-rabbitmq-jpa/domain/redis/src/main/kotlin/sample/domain/cache/repository/PasswordResetRequestRepository.java
package sample.domain.cache.repository;

import org.springframework.data.repository.CrudRepository;
import sample.domain.cache.model.PasswordResetRequestCache;

import java.util.Optional;

public interface PasswordResetRequestRepository extends CrudRepository<PasswordResetRequestCache, String> {
    Optional<PasswordResetRequestCache> findOneByCode(String code);
}

File: 2021-rabbitmq-jpa/domain/redis/src/main/kotlin/sample/domain/cache/repository/PasswordResetRequestRepository.kt
package sample.domain.cache.repository

import org.springframework.data.repository.CrudRepository
import sample.domain.cache.model.PasswordResetRequestCache
import java.util.*

interface PasswordResetRequestRepository : CrudRepository<PasswordResetRequestCache, String> {
    fun findOneByCode(code: String): Optional<PasswordResetRequestCache>
}
